nix/pkgs/haskell/haskell.nix

```diff
@@ -20,6 +20,7 @@ let
   r-packages = with rPackages; [ R tidyverse dplyr stringr MASS plotly shiny shinyjs purrr ];
   project = haskell-nix.cabalProject' ({ pkgs, ... }: {
     inherit compiler-nix-name;
+    evalPackages = pkgs.buildPackages.buildPackages;
```

Comment:
comment, please! Also I don't think we want this: this makes it not possible to evaluate things on a different system from the build one, which we used to be able to do. So should we not actually use `evalPackages` here?

Reply:
Indeed, isn't that the default? or is the issue that the default is broken?

Reply:
`evalPackages` default is `pkgs.buildPackages`. But apparently this not enough, probably because:
```
nix eval -f ./default.nix --arg crossSystem '{ config = "x86_64-w64-mingw32"; libc = "msvcrt"; }' plutus.haskell.project.pkgs.buildPackages.targetPlatform.system
"x86_64-windows"
```
while with `pkgs.buildPackages.buildPackages`
```
nix eval -f ./default.nix --arg crossSystem '{ config = "x86_64-w64-mingw32"; libc = "msvcrt"; }' plutus.haskell.project.pkgs.buildPackages.buildPackages.targetPlatform.system
"x86_64-linux"
```
But not sure why exactly. Maybe @hamishmack ?

Reply:
Oh that is a good point.  Just using `.buildPackages` has caused problems in the past with caching.  IIRC anything that used python (including some surprising things like git) would include a dependency on `gdb` (I think it was `gdb`), but `buildPackages.gdb` would be the cross compiler `gdb` (not the native one).  The result was that git would recompile.

Reply:
See https://github.com/input-output-hk/haskell.nix/pull/1739